src/main/java/ch/njol/skript/events/EvtEntity.java

```diff
@@ -24,24 +24,28 @@
 import org.bukkit.event.Event;
 import org.bukkit.event.entity.EntityDeathEvent;
 import org.bukkit.event.entity.EntitySpawnEvent;
+import org.bukkit.event.entity.PlayerDeathEvent;
 import org.eclipse.jdt.annotation.Nullable;
 
 import ch.njol.skript.Skript;
 import ch.njol.skript.entity.EntityData;
 import ch.njol.skript.lang.Literal;
 import ch.njol.skript.lang.SkriptEvent;
 import ch.njol.skript.lang.SkriptParser.ParseResult;
-import ch.njol.skript.log.ErrorQuality;
+import ch.njol.skript.lang.util.SimpleEvent;
 import ch.njol.skript.registrations.Classes;
 import ch.njol.util.StringUtils;
 
-/**
- * @author Peter Güttinger
- */
-@SuppressWarnings("unchecked")
 public final class EvtEntity extends SkriptEvent {
-	
+
 	static {
+		// Must be registered before EntityDeathEvent.
+		Skript.registerEvent("Player Death", SimpleEvent.class, PlayerDeathEvent.class, "death of [a] player", "player death")
+				.description("Called when a player dies.")
+				.examples("on death of player:",
```

Comment:
Should this be `SimpleEvent.class` or `EvtEntity.class`
additionally would this be worth adding as a `.noDoc()`?
I don't see much use in it being documented when it use to be apart of the `EntityDeathEvent` event
also would `on death of an player` run for the `EntityDeathEvent`?

Reply:
It's supposed to be a SimpleEvent, it's only present in EvtEntity class because it must be registered before the other events to be able to override it. It should never parse `on death of player` as the EntityDeathEvent.

Reply:
`of player` makes sense but `an player` is what concerns me as the event only cares for `[a] player` someone making a dumb little mistake as `on death of an player` should got with `EntityDeathEvent` no?

Reply:
Good catch. I originally had it with `[a[n]]` but it doesn't sound grammarly correct, but it must allow for `an` due to entitydata having `@an` defined in the default.lang.

Reply:
Users can still do `on death of a player, pig or wither` which is fine. That'll work, but the syntaxes that rely on PlayerDeathEvent will error, and that's something that cannot be fixed.

Reply:
yeah that's fine most people don't really put multiple entities in a single event and if they do they won't have a need for any syntax explicitly for players